pennylane/devices/qutrit_mixed/initialize_state.py

```diff
@@ -0,0 +1,119 @@
+
+    Args:
+        wires (Union[Wires, Iterable]): The wires to be present in the initial state
+        prep_operation (Optional[StatePrepBase]): An operation to prepare the initial state
+        like (Optional[str]): The machine learning interface used to create the initial state.
+            Defaults to None
+
+    Returns:
+        array: The initial state of a circuit
+    """
+    if isinstance(wires, Wires):
+        wires = wires.tolist()
+
+    num_wires = len(wires)
+
+    if not prep_operation:
+        rho = _create_basis_state(num_wires, 0)
+
+    elif isinstance(prep_operation, QutritBasisState):
+        rho = _apply_basis_state(prep_operation.parameters[0], wires)
+
+    elif isinstance(prep_operation, StatePrepBase):
+        rho = _apply_state_vector(prep_operation.state_vector(wire_order=list(wires)), num_wires)
+
+    # TODO: add instance for prep_operations as added
+
+    return qml.math.asarray(rho, like=like)
+
+
+def _apply_state_vector(state, num_wires):  # function is easy to abstract for qudit
+    """Initialize the internal state in a specified pure state.
+
+    Args:
+        state (array[complex]): normalized input state of length
+            ``3**len(wires)``
+        num_wires (int): number of wires that get initialized in the state
+
+    Returns:
+        array[complex]: complex array of shape ``[3] * (2 * num_wires)``
+        representing the density matrix of this state.
+    """
+
+    # Initialize the entire wires with the state
+    rho = qml.math.outer(state, qml.math.conj(state))
+    return qml.math.reshape(rho, [3] * 2 * num_wires)
+
+
+def _apply_basis_state(state, wires):  # function is easy to abstract for qudit
+    """Returns initial state for a specified computational basis state.
+
+    Args:
+        state (array[int]): computational basis state of shape ``(wires,)``
+            consisting of 0s, 1s and 2s.
+        wires (Iterable[int]): wires that the provided computational state should be initialized on
+
+    Returns:
+        array[complex]: complex array of shape ``[3] * (2 * num_wires)``
+        representing the density matrix of this basis state.
+    """
+    num_wires = len(wires)
+
+    # get computational basis state number
+    basis_states = qudit_dim ** (num_wires - 1 - np.array(wires))
+    num = int(qml.math.dot(state, basis_states))
+
+    return _create_basis_state(num_wires, num)
+
+
+def _create_basis_state(num_wires, index):  # function is easy to abstract for qudit
+    """Return the density matrix representing a computational basis state over all wires.
+
+    Args:
+        num_wires (int): number of wires to initialize
+        index (int): integer representing the computational basis state.
+
+    Returns:
+        array[complex]: complex array of shape ``[3] * (2 * num_wires)``
+        representing the density matrix of the basis state.
+    """
+    rho = np.zeros((3**num_wires, 3**num_wires))
```

Comment:
```suggestion
    rho = qml.math.zeros((3**num_wires, 3**num_wires))
```

Reply:
It is done with NumPy in default qutrit. I don't think there will be a difference since qml.math.asarray(rho, like=like) is called. Should I change it?

Reply:
I think it's probably best to change it. As far as the structure is concerned, `default.qubit` will be a better reference than `default.qutrit` since the latter uses the old device API. `default.qubit` only uses `qml.math`. The only place I can think of that uses NumPy is when we generate samples using the probability vector in `pennylane/devices/qubit/sampling.py`.

Reply:
oops  I meant `default.qubit`, I was looking at that since it uses the new device API, so many qubits and qutrits has turned me around.

Reply:
Looking at this again, I'm impartial to whether `numpy` or `qml.math` gets used.

Reply:
I'm going to change to qml.math, there is a test to make sure that the right type is returned anyway.